src/com/sahokia/kos/handmadecollections/MyHashSet.java

```diff
@@ -0,0 +1,100 @@
+package com.sahokia.kos.handmadecollections;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+
+public class MyHashSet<T> implements Iterable<T> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private LinkedList<T>[] buckets;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashSet() {
+        buckets = new LinkedList[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private int hash(T element) {
+        return (element == null) ? 0 : Math.abs(element.hashCode() % buckets.length);
+    }
+
+    public boolean put(T element) {
+        int index = hash(element);
+        if (buckets[index] == null) {
+            buckets[index] = new LinkedList<>();
+        }
+        if (!buckets[index].contains(element)) {
+            buckets[index].add(element);
+            size++;
+            if (size >= LOAD_FACTOR * buckets.length) {
+                resize();
+            }
+            return true;
+        }
+        return false;
+    }
+
+    public boolean contains(T element) {
+        int index = hash(element);
+        return buckets[index] != null && buckets[index].contains(element);
+    }
+
+    public boolean delete(T element) {
+        int index = hash(element);
+        if (buckets[index] != null && buckets[index].remove(element)) {
+            size--;
+            return true;
+        }
+        return false;
+    }
+
+    @SuppressWarnings("unchecked")
```

Comment:
What does this annotation? Why do we need it here?

Reply:
Just used to avoid warnigs about unchecked type cast) but we don't have actually have problems or any risks with it because the collection is parameterized itself and we cannot put not parameterized element inside. If this annotation is not commonly used, I remove it